### PR TITLE
fixing possible NPE in Apache Camel instrumentation

### LINKED ISSUE #2243: Camel instrumentation can suffer from NPE in AWS parent propagation scenario
**Describe the bug**
If SQS message is produced without propagation header (AWSTraceHeader) set, consuming by Apache Camel instrumentation can cause NPE.

**Steps to reproduce**
- send message without propagation (eg by not instrumented code)
- consume with Camel
- parent extraction (io.opentelemetry.javaagent.instrumentation.apachecamel.CamelPropagationUtil#extractAwsPropagationParent) can suffer from NPE due to NULL value

**What did you expect to see?**
- propagation should work

**What did you see instead?**

**What version are you using?**
- master

**Environment**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by kubawach**
Fixes #2243

Chance of NPE in case of SQS parent propagation from a message without AWSTraceHeader (eg not instrumented producer).

### CODE REVIEW

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java [hunk 1]
```diff
@@ -58,8 +58,12 @@ static void injectParent(Context context, final Map<String, Object> exchangeHead
...
 
     @Override
-    public String get(Map<String, Object> map, String s) {
-      return (map.containsKey(s) ? map.get(s).toString() : null);
+    public String get(Map<String, Object> map, String key) {
+      return (hasValidValue(map, key) ? map.get(key).toString() : null);
+    }
+
+    private boolean hasValidValue(Map<String, Object> map, String key) {
+      return (map.containsKey(key) && (map.get(key) != null));
```
Comment (breedx-splk) [inline instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:66]: I think that calling `containsKey()` prior to `get()` is clunky.  This `get()` is also repeated in the caller just a few lines above.  I suggest combining these methods into a simple 2-liner:
```
Object value = map.get(key);
return value == null : null : value.toString();
```

**Review by breedx-splk — CHANGES_REQUESTED** [review]
This looks good to me. Had one small suggestion around readability.  👍

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java (hunk 1 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:66]: Makes sense :)

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java [hunk 2]
```diff
@@ -58,8 +58,9 @@ static void injectParent(Context context, final Map<String, Object> exchangeHead
     }
 
     @Override
-    public String get(Map<String, Object> map, String s) {
-      return (map.containsKey(s) ? map.get(s).toString() : null);
+    public String get(Map<String, Object> map, String key) {
+      Object value = map.get(key);
+      return (value == null ? null : map.get(key).toString());
```
Comment (breedx-splk) [inline instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:63]: You still have the double get!  :). You can just say `value.toString()` here.

Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java]: Too fast, too furious 🙈 But in reality I probably should stop working after 6 PM 😬

**Review by breedx-splk — APPROVED** [review]
Looks good to me, thanks!

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java [hunk 3]
```diff
@@ -58,8 +58,9 @@ static void injectParent(Context context, final Map<String, Object> exchangeHead
     }
 
     @Override
-    public String get(Map<String, Object> map, String s) {
-      return (map.containsKey(s) ? map.get(s).toString() : null);
+    public String get(Map<String, Object> map, String key) {
+      Object value = map.get(key);
+      return (value == null ? null : value.toString());
```
Comment (anuraaga) [inline instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:63]: I agree this new style is cleaner but my eyes aren't being able to see the NPE fix - is the map bring mutated between containsKey and get previously?

**Review by anuraaga — APPROVED** [review]

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java (hunk 3 above)
Comment (iNikem) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:63]: I understand that map has an explicit value `null` assigned to the given key.

**Review by iNikem — APPROVED** [review]

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java (hunk 3 above)
Comment (kubawach) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:63]: Nikita is right, previously we only checked if there is a key/value mapping and dereferenced value if so.

Comment (anuraaga) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java:63]: Got it, thanks!

**Pull request merged**